modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/kill/LocalToClusterKillHandlerWrapper.java

```diff
@@ -0,0 +1,149 @@
+        return localHandler.cancelAsync(operationId)
+                .thenCompose((result) -> {
+                    if (Boolean.TRUE.equals(result)) {
+                        return CompletableFuture.completedFuture(Boolean.TRUE);
+                    }
+
+                    CancelOperationRequest request = FACTORY.cancelOperationRequest()
+                            .operationId(operationId)
+                            .typeId(type.id())
+                            .build();
+
+                    return broadcastCancel(request);
+                });
+    }
+
+    private CompletableFuture<Boolean> broadcastCancel(CancelOperationRequest request) {
+        CompletableFuture<Boolean> result = new CompletableFuture<>();
+
+        CompletableFuture<?>[] futures = logicalTopologyService.localLogicalTopology().nodes()
+                .stream()
+                .filter(node -> !node.name().equals(localNodeName))
+                .map(node -> messageService.invoke(node, request, RESPONSE_TIMEOUT_MS)
+                        .thenAccept(msg -> {
+                            CancelOperationResponse response = (CancelOperationResponse) msg;
+                            Throwable remoteErr = response.error();
+
+                            if (remoteErr != null) {
+                                LOG.warn("Remote node returned an error while canceling the operation "
+                                                + "[operationId={}, typeId={}, node={}].", remoteErr,
+                                        request.operationId(), request.typeId(), node.name());
+                            }
+
+                            if (response.error() != null) {
+                                throw new CompletionException(response.error());
+                            }
+
+                            Boolean res = response.result();
+
+                            if (Boolean.TRUE.equals(res)) {
+                                result.complete(true);
+                            }
+                        })
+                        .whenComplete((ignore, err) -> {
+                            if (err != null) {
+                                LOG.warn("Failed to send a request to cancel the operation to the remote node "
+                                        + "[operationId={}, typeId={}, node={}].", err,
+                                        request.operationId(), request.typeId(), node.name());
+                            }
+                        })
+                )
+                .toArray(CompletableFuture[]::new);
+
+        CompletableFuture.allOf(futures)
+                .whenComplete((unused, throwable) -> {
+                    if (result.isDone()) {
+                        return;
+                    }
+
+                    if (throwable == null) {
+                        result.complete(Boolean.FALSE);
+
+                        return;
+                    }
+
+                    result.completeExceptionally(throwable);
```

Comment:
not sure about contract of exceptional `CompletableFuture.allOf`, but throwable will be the first thrown exception, while in fact several request may fail with different errors. I suggest to link all exception in chain of suppressed exceptions (see `org.apache.ignite.internal.sql.engine.exec.ExecutionServiceImpl#deriveExceptionFromListOfFutures` for example). It worth to add test on that matter

Reply:
Thanks, fixed.
Added this case in test `KillCommandHandlerTest.testCancelFinishesWithErrorOnRemoteNode`